tools/matrix-bot/logger.ts

```diff
@@ -22,24 +17,49 @@ export class MessagesLogger {
     if (!eventId) {
       return;
     }
-    if (!msg.includes("graypaper.fluffylabs.dev")) {
+    const gpLinks = findLinks(msg);
+    if (!gpLinks.length) {
       return;
     }
 
+    // if there are several links in the message, find the link with the latest version
+    let versionName: string | undefined;
+
+    if (gpLinks.length === 1) {
+      versionName = parseLink(gpLinks[0], this.meta)?.versionName;
+    } else {
+      versionName = gpLinks
+        .map((link) => parseLink(link, this.meta))
+        .filter(Boolean)
+        .map((link) => link?.versionName)
+        .sort()
+        .pop();
+    }
+
+    if (!versionName) {
+      return;
+    }
+
+    const majorVersion = versionName.replace(/^(\d+\.\d+\.)\d+$/, "$1x");
+    const outputFilename = `output/messages-${majorVersion}.json`;
+
     const link = this.generatePermalink(eventId);
+    const json = JSON.stringify({
+      date,
+      sender,
+      link,
+      msg,
+    });
 
-    this.file.write(
-      JSON.stringify({
-        date,
-        sender,
-        link,
-        msg,
-      }),
-    );
-    this.file.write(",\n");
-  }
+    try {
+      const messages = require(path.resolve(outputFilename));
+      messages.push(JSON.parse(json));
+      writeFileSync(outputFilename, JSON.stringify(messages));
+    } catch (e) {
+      writeFileSync(outputFilename, JSON.stringify([JSON.parse(json)]));
+    }
```

Comment:
```suggestion
    try {
      const messages = require(path.resolve(outputFilename));
      messages.push(json);
      writeFileSync(outputFilename, JSON.stringify(messages));
    } catch (e) {
      writeFileSync(outputFilename, JSON.stringify([json]));
    }
```

seems that in both branches you are parsing the json that was just freshly stringified. Is there reason for that that I'm missing?

Reply:
Also I think the `try` block could be made a bit more minimal:
```ts
let messages = [];
try {
  messages = require(path.resolve(outputFilename));
} catch(e) {}
messages.push(json);
writeFileSync(outputFilename, JSON.stringify(messages));
```

Reply:
Thirdly, AFAIK `require` will cache the result, so the JSON file will only be read once and subsequent `requires` will just return a cached object.

Not sure if it was tested, but my gut feeling is that this code will keep writing the same file over and over again (aka it only works for the first message after being run).

Reply:
its been tested and wrote all messages fine. your point does look legit but for some reason it still works with require. in any case im changing it to `readFileSync` to be safe